### PR TITLE
Use JSON-RPC for REPL-extension comm

### PR DESCRIPTION
**Pull request by davidanthoff**
Improves #1203.

Major changes:
- Uses JSON RPC
- We use the same implementation in the REPL that we use in LS on the Julia side (I moved the code into a package)
- On the TypeScript side we use the vscode-jsonrpc package
- We use only one named pipe instead of multiple in the previous version
- All the weird message format hacks are gone :)
- We vendor JSON.jl into the REPL process, which will be super useful for lots of other things as well
- I'm starting to move all the Julia packages that we ship into the folder `scripts/packages`, right now they are all over the place.

Once we have this merged, I'll also use this for the debugger communication.

I don't yet have the kind of callback implementation that Juno has. The main reason is that I realized that won't work in the simple form for LS, but I want the JSONRPC package to work for that scenario as well. I'll open another issue to discuss the specific problem.

### CODE REVIEW

**Comment by davidanthoff** [comment]
@ZacLN: just fixed another small problem here, but now it should be ready for review.

File: scripts/terminalserver/terminalserver.jl [hunk 1]
```diff
@@ -1,59 +1,55 @@
...
 function ends_with_semicolon(x)
     return REPL.ends_with_semicolon(split(x,'\n',keepempty = false)[end])
 end
 
+repl_conn = connect(repl_pipename)
+
+conn_endpoint = JSONRPC.JSONRPCEndpoint(repl_conn, repl_conn)
+
+function sendDisplayMsg(kind, data)
+    JSONRPC.send_notification(conn_endpoint, "display", Dict{String,String}("kind"=>kind, "data"=>data))
```
Comment (ZacLN) [inline scripts/terminalserver/terminalserver.jl:34]: Is a more specific notification name needed?

File: src/repl.ts [hunk 2]
```diff
@@ -559,62 +570,40 @@ function processMsg(cmd, payload) {
...
-                let msg_len = parseInt(msg_len_as_string);
-                if (accumulatingBuffer.length >= cmd.length + msg_len_as_string.length + 2 + msg_len) {
-                    let payload = s.substring(index_of_sep_2 + 1);
-                    if (accumulatingBuffer.length > cmd.length + msg_len_as_string.length + 2 + msg_len) {
-                        accumulatingBuffer = Buffer.from(accumulatingBuffer.slice(cmd.length + msg_len_as_string.length + 2 + msg_len + 1));
-                    }
-                    else {
-                        accumulatingBuffer = Buffer.alloc(0);
-                    }
+const notifyTypeDisplay = new rpc.NotificationType<{kind: string, data: any}, void>('display');
```
Comment (ZacLN) [inline src/repl.ts:578]: As above?

**Comment by davidanthoff** [comment]
I think `display` is fine for now, it really is the equivalent of `Base.display`. But we can also revisit later and change.

> is there a reason we don't pass NUM_THREADS here?

That looks like a bug/oversight to me, I think we should pass it there as well.

**Pull request merged**